src/uct/sm/knem/knem_ep.c

```diff
@@ -40,23 +40,35 @@ UCS_CLASS_DEFINE_DELETE_FUNC(uct_knem_ep_t, uct_ep_t);
         return UCS_OK;                                  \
     }
 
-static inline ucs_status_t uct_knem_rma(uct_ep_h tl_ep, const void *buffer,
-                                        size_t length,  uint64_t remote_addr,
-                                        uct_knem_key_t *key, int write)
+static inline ucs_status_t uct_knem_rma(uct_ep_h tl_ep, const uct_iov_t *iov,
+                                        size_t iovcnt, size_t *length,
+                                        uint64_t remote_addr, uct_knem_key_t *key,
+                                        int write)
 {
     struct knem_cmd_inline_copy icopy;
-    struct knem_cmd_param_iovec knem_iov[1];
+    struct knem_cmd_param_iovec knem_iov[iovcnt];
```

Comment:
The user must provide the value less than max_iov. In case of error it can be LLONG_MAX.
Would you like me to use alloca?